benchadapt/python/benchadapt/machine_info.py

```diff
@@ -71,11 +73,48 @@ def python_info():
 
 
 def github_info():
+    """Attempts to inspect a locally cloned repository for git information that can be
+    posted to the "github" key when creating a run.
+
+    It's recommended to NOT use this function, and instead manually supply the commit
+    and repository (in the form "org/repo"). When doing so, ignore the "branch" key
+    and only supply the "pr_number" (or leave it None if it's a commit to the default
+    branch).
+    """
     commit = _exec_command(["git", "rev-parse", "HEAD"])
-    repository = _exec_command(["git", "remote", "get-url", "origin"])
+    if not commit:
+        warnings.warn(
+            "error in github_info(): probably not in a git repo; returning {}"
+        )
+        return {}
+
+    branches = _exec_command(["git", "branch", "-vv"])
+    if "* (HEAD detached" in branches:
+        remote = "origin"
+        branch = "<DETATCHED BRANCH>"
+        warnings.warn(f"error in github_info(): detached HEAD; returning {branch=}")
```

Comment:
This is the case for most build systems (they run on a detached HEAD that's the hypothetical merge commit) so that's another reason to recommend that most systems should just supply the PR number.

Reply:
Ha, indeed! I forgot about this too

Reply:
They'll usually make git info available in env vars though (e.g. [jenkins](https://plugins.jenkins.io/git/#plugin-content-branch-variables)). I wonder if the particular env vars are shared enough across build systems such that we could just check if they're populated? Or make our own as a way for users to pass us the info.

Reply:
What I mean is it's the same amount of work—if not more—to have runners get the branch env var and format it `org:branch` than it is to have runners get the PR number env var. So we're actually not saving any configuration if we recommend the "autoscrape from git" method.

It is interesting to think about autoscraping this stuff from the environment, but yeah we'll probably need to deal with at least jenkins, github, buildkite, and circleci, if not more. I think they mostly have different env var names. So that could eventually be more of a pain than it sounds

Reply:
Yeah, that would be a lot of work unless everybody already uses the same env var names. 

It may still be worth it to make our own env vars to check, though—otherwise most implementations will need to make them themselves anyway and pass them through to the adapter or whatever is making this JSON.

Reply:
Yeah that's a great point. I'll defer that to the adapter work.